src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.SetUnixFileMode.NonLinux.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        private int ChModNoFollowLink(string path, int mode)
+        {
+            EnsureCachesInitialized(path);
+
+            if (!EntryExists)
+                FileSystemInfo.ThrowNotFound(path);
```

Comment:
We shouldn't need `EnsureCachesInitialized + EntryExists` as `LChMod` will also tell us when the entry doesn't exist. These calls are included to have the `DirectoryNotFoundException` mapping that gets performed by `ThrowNotFound`.

https://github.com/dotnet/runtime/pull/70661 moves the mapping so it will also be used when checking the return value of `LChMod`. With that change, we can eliminate `EnsureCachesInitialized + EntryExists` here.